### PR TITLE
[bug fix] Render corner cases

### PR DESCRIPTION
**Pull request by dturn**
**What are you trying to accomplish with this PR?**

1. Disable rendering by default in DeployTask.

2. Fix bugs with the render task:

(a) Thor defaults the value of a flag with no values passes as the flag name, e.g. `--current-sha ` sets the value of of `options['current-sha'] == "current-sha"`
(b) `krane render --std-in` doesn't actually render

**How is this accomplished?**
1. flip the default to false
2.a set a lazy default
2.b pass information `OptionsHelper` that the file name for data from stdin needs to end in `.erb`

**What could go wrong?**

Not much with the bug fixes. The default flip is part of the 1.0 breaking changes.

### CODE REVIEW

File: lib/krane/render_task.rb [hunk 1]
```diff
@@ -91,6 +91,10 @@ def validate_configuration(template_sets)
       if @filenames.blank?
         errors << "filenames must be set"
       end
+
+      if !@current_sha.nil? && @current_sha.empty?
+        errors << "current-sha is optional but can not be blank"
+      end
```
Comment (dirceu) [inline lib/krane/render_task.rb]: This (and the `@filenames` check right before it) might be better suited to a `TaskConfigValidator` subclass, if we want to follow the same pattern we used with the others tasks; I think it's totally fine to move that in another PR though.

File: test/test_helper.rb [hunk 2]
```diff
@@ -227,9 +227,9 @@ def task_config(context: KubeclientHelper::TEST_CONTEXT, namespace: @namespace,
       @task_config ||= Krane::TaskConfig.new(context, namespace, logger)
     end
 
-    def krane_black_box(command, args = "")
+    def krane_black_box(command, args = "", stdin: nil)
```
Comment (timothysmith0609) [inline test/test_helper.rb]: :heart:

File: lib/krane/render_task.rb (hunk 1 above)
Comment (dturn) [reply lib/krane/render_task.rb]: Tracking in #637
